app/main.py

```diff
@@ -1,3 +1,69 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        else:
+            return Distance(
+                km=self.km + other
+            )
+
+    def __iadd__(self, other: Distance | int) -> Distance:
```

Comment:
The type hint `Distance | int` is only supported in Python 3.10 and later. If you need compatibility with earlier versions, use `Union[Distance, int]` from the `typing` module.